lib/printers/json.js

```diff
@@ -0,0 +1,26 @@
+const Linter = require('../index');
+
+class JsonPrinter {
+  constructor(options = {}) {
+    this.options = options;
+    this.console = options.console || console;
+  }
+
+  print(errors) {
+    let filteredErrors = {};
+    for (let filePath of Object.keys(errors)) {
+      let fileErrors = errors[filePath] || [];
+
+      let errorsFiltered = fileErrors.filter(error => error.severity === Linter.ERROR_SEVERITY);
+      let warnings = this.options.quiet
+        ? []
+        : fileErrors.filter(error => error.severity === Linter.WARNING_SEVERITY);
+
+      filteredErrors[filePath] = errorsFiltered.concat(warnings);
+    }
+
+    this.console.log(JSON.stringify(filteredErrors, null, 2));
+  }
```

Comment:
Should some of this get extracted to a helper method? Seems like both `pretty` and `json` want most of this same logic.

Reply:
I guess alternatively, `pretty` could extend from `json` (since it can use the well known JSON structure to emit the pretty messages).

Reply:
yeah, idk, the pretty implementation also counts the number of issues, so I figured the deduplication wasn't worth it for now. especially since the json/pretty printers are currently mutually exclusive.

Reply:
Gotcha, seems totally fine to me.